### PR TITLE
Layer Panel: Accessibility improvements

### LINKED ISSUE #4432: A11Y: Design > Layers
## Solution to be implemented

1. Hide entire layer panel for screen readers by using `aria-hidden="true"`
1. Display layer manipulation shortcuts in the keyboard shortcut dialog:
  a. <kbd>⌘</kbd><kbd>↑</kbd> - move element forward
  b. <kbd>⌘</kbd><kbd>↓</kbd> - move element backward
  c. <kbd>⌘</kbd><kbd>⇧</kbd><kbd>↑</kbd> - move element to front
  d. <kbd>⌘</kbd><kbd>⇧</kbd><kbd>↓</kbd> - move element to back
#### 1. Navigating into Layers section blocked after expansion.

- User enables VoiceOver while Layers is collapsed. 
- User expands Layers via the arrow toggle button. 
- Expected: User selects "next item" in screen reader and the layers list group is selected.
- Actual: User is unable to navigate beyond the toggle button into the list group of layers via "next item" VoiceOver navigation. 
  - User's VoiceOver navigation actually becomes trapped between the "Layers" label and the toggle button.

#### 2. Layer items in list group are announced twice.

- User enables VoiceOver while Layers is expanded.
- User navigates into the Layers list group.
- Expected: Each layer is announced once with its layer contents (e.g. text, image, video).
- Actual: Each layer is preceded by a blank group (e.g. "group 1 of 16" with 8 layers).

## Additional Context

- Plugin Version: 1.0.0-beta.2+fd68af7
- Operating System: macOS 10.15.6
- Browser: Chrome 85

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
## Summary

1. Hides layer panel from screen readers
1. Adds layer manipulation shortcuts to shortcut modal.

Please see #4432 for arguments.

## Testing Instructions

1. Open keyboard shortcut modal
1. See new shortcuts listed

---

Fixes #4432

### CODE REVIEW

File: assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js
Comment (miina) [inline assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: Hmm, just `drag` doesn't seem precise enough to reflect that it happens when resizing as well. Thoughts?

File: assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js
Comment (miina) [inline assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: I wonder if the word `layer` is even necessary for this context -- since it has "Element" as the group title. Might be confusing, especially with Voice-Over. WDYT?

File: assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js
Comment (barklund) [reply assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: It was mostly for visual reasons, as the previous version looked crappy IMO. The shortcuts take more space now with the longer shortcuts for move elements in layer ordering, so there's less room for the descriptions.

@o-fernandez, what do you think?

Old text before:
Screen Shot 2020-09-17 at 09 07 40


Old text now:
Screen Shot 2020-09-17 at 09 06 21


New text:
Screen Shot 2020-09-17 at 09 06 50

File: assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js
Comment (barklund) [reply assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: It is to specify, that it's not actually moving the element itself, but the layer the element is in. I'm open to other copy. @o-fernandez, what do you think?

**Review by miina — APPROVED** [review]
LGTM, pending the text confirmation.

File: assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js
Comment (barklund) [reply assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: And yes, we could reduce the space used for the shortcuts by having separate entries for move element up/down and front/back, but then the list becomes longer and that also looked back - especially compared to the left hand side:

Separate entries:
Screen Shot 2020-09-17 at 09 34 43

Compact entries:
Screen Shot 2020-09-17 at 09 32 27

Comment (mariano-formidable) [reply assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: I think the compact version loooks good to me.  

But, we could also take the layer shortcuts out of the `Element` section, and create a new `Layer` section that only has those shortcuts.  Then you could update the shortcut render logic so the left column has `Basics`, `Text`, `Layers`  shortcuts and the right side only has `Element`.  Maybe we can add a `col` prop to each section with values left or right (if it's missing it assumes left), and then the columns could be easily rearranged by changing the col property per section 🤔.

File: assets/src/edit-story/components/keyboardShortcutsMenu/constants.js
Comment (mariano-formidable) [inline assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: When I first loaded up the shortcuts window, this is how the up/down arrows are rendering on my machine (Mac, latest Chrome and latest Firefox):
image

It looks like WordPress is auto replacing the up/down arrow with emoji-images when it loads this script: `wp-emoji.js?ver=5.5.1`.  When I use chrome to block the url to that script, preventing WP from loading it, then the short cuts appear as we would expect:

image

Not sure if this is an issue, but just wanted to call it out.

**Review by mariano-formidable — APPROVED** [review]
Works as expected, I called out a couple of things for you to consider, but I don't think they're blockers.

File: assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js
Comment (o-fernandez) [reply assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: I like compact. In terms of language 
**"Disable snapping while moving"** 
seems more natural to me.

File: assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js
Comment (o-fernandez) [reply assets/src/edit-story/components/keyboardShortcutsMenu/getKeyboardShortcuts.js]: I think that both: 
**Move forward or backward**
**Move to front or back**

Work well, so we can drop layer.

File: assets/src/edit-story/components/keyboardShortcutsMenu/constants.js
Comment (barklund) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: Oh, that's weird. I'm not seeing that - is that some plugin or built-in behavior? @swissspidy, do you know?

And @o-fernandez, is this an issue?

Comment (o-fernandez) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: It'd be great to not have that happen, but if it comes between shipping the feature with that issue or delaying, I'd ship and fix later. @swissspidy any thoughts on what may be happening?

**Comment by csossi** [comment]
Verified in QA

File: assets/src/edit-story/components/keyboardShortcutsMenu/constants.js
Comment (swissspidy) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: @barklund @o-fernandez  That's some built-in behavior in WordPress, yes. WordPress uses the [Twemoji](https://twemoji.twitter.com/) library to override emoji with images under some circumstances.

@barklund You can add a `wp-exclude-emoji` class name to prevent this from happening in this case.

Comment (swissspidy) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: Oh, looks like that workaround only works on non-element nodes. https://wordpress.org/support/article/emoji/#converting-to-images

Comment (swissspidy) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: An alternative would be to use small SVGs with proper aria-labels.

Comment (barklund) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: Another alternative is to use "regular" (non-emoji) unicode arrows, but it doesn't look as good IMHO:

With `UPWARDS/DOWNWARDS ARROW`:
Screen Shot 2020-09-18 at 11 01 52

With `UPWARDS/DOWNWARDS TRIANGLE-HEADED ARROW`:
Screen Shot 2020-09-18 at 11 05 42

In summary, I see the following options:

1. Use a small SVG (will take a little time, but not much - will make it more complex though)
1. Use one of the Unicode arrow sets above 
1. F*ck it, ship it.

@o-fernandez, your call!

Comment (o-fernandez) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: I think 1 or 3.
I don't really like how 2 looks, particularly if we keep Shift as it is now.

Comment (barklund) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: Shipped as is.

Comment (samitron7) [reply assets/src/edit-story/components/keyboardShortcutsMenu/constants.js:39]: Just no...

**Pull request merged**
